src/Model/DoctrineAwareManagerInterface.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Model;
+
+/**
+ * @author Jordi Sala <jordism91@gmail.com>
+ */
+interface DoctrineAwareManagerInterface
```

Comment:
WDYT about this interface @greg0ire ?

Reply:
I find it weird that it seems to be used only in tests, and I have a bad feeling about this dependency on Doctrine. Maybe it is a necessary evil

Reply:
the only alternative to me is to add those methods to the modelManager interface on next major, and use the dirty trick I had before on tests. I guess this is not the only place where this happens.

My plan if we accept this new interface is to provide PRs to the persistence bundles to implement this interface.

There is always another alternative, but implies we remove the if check that uses the metadata. If I remove that, I won’t need the interface.

None of those solutions seems perfect to me. To be honest, the dirty trick didnt seem a super bad option, because it gives some space to do it better in the future, introducing a new interface gives more constraints.

Reply:
Yeah I can't think of anything better either. @sonata-project/contributors, should we accept this interface or remove the code that uses it?

Reply:
I will try what happens with and without that check first